src/app/visit/visit.cpp

```diff
@@ -0,0 +1,59 @@
+#include "visit.hpp"
+#include "../doctor/doctor.hpp"
+
+std::set<Visit *> Visit::visitExtent;
+
+Visit::Visit()
+{
+    visitExtent.insert(this);
+}
+
+Visit::~Visit()
+{
+    removeFromExtent(this);
+}
+
+std::shared_ptr<Visit> Visit::createVisit(std::shared_ptr<Doctor> doc)
```

Comment:
I think, that we should have more arguments(Date/Hour, Room, Patient etc.) or at least provide overloads.

In this approach we just create an empty visit and then have to edit it somewhere in the code, without means of verification if a constructed objects are fully valid.
Having it done by calling a specific constructor inside (which should also be overloaded), we could establish a well defined invariant in the constructor.